### PR TITLE
Niloofar/ Submenu component

### PR DESCRIPTION
**Pull request by niloofar-deriv**
https://github.com/deriv-com/ui/assets/93518187/4ce5d14c-67c0-4abd-a02d-0ca654a82124

### CODE REVIEW

**Review by ali-hosseini-deriv — CHANGES_REQUESTED** [review]
there is a padding or margin between the icon and the text in production. dont we need that?

**Review by ali-hosseini-deriv — APPROVED** [review]

File: src/components/AppLayout/Submenu /index.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * @param {ReactNode} props.icon - The icon displayed in the button that toggles the submenu.
+ * @param {string} props.label - The label text displayed next to the icon in the toggle button.
+ * @param {string} [props.labelSize="md"] - The size of the label text, defaults to "md".
+ * @param {ReactNode} props.submenuContent - The content displayed inside the submenu when it is open.
+ * @param {string} [props.submenuClassName] - Optional custom class name for styling the submenu container.
+ * @param {string} [props.className] - Optional custom class name for styling the toggle button.
+ * @param {ReactNode} props.children - The children nodes provided to the submenu panel, which are displayed below the submenuContent.
+ * @returns {JSX.Element} The rendered Submenu component with toggle functionality.
+ */
+export const Submenu = ({
```
Comment (shayan-deriv) [inline src/components/AppLayout/Submenu /index.tsx:32]:
```suggestion
export const SubMenu = ({
```
also for the rest please 🙏

**Review by farrah-deriv — APPROVED** [review]

**Comment by niloofar-deriv** [comment]
> there is a padding or margin between the icon and the text in production. dont we need that?

Base on our discussion since this is a ui library I don't want to hard code these sizes so I added some props and customer would have the freedom to set the sizes

File: src/components/AppLayout/Submenu /__tests__/Submenu.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        render(
+            <Submenu
+                icon={<span>Icon</span>}
+                label="Account settings"
+                submenuContent={<span>Close</span>}
+            />,
+        );
+
+        expect(screen.queryByText("Close")).not.toBeInTheDocument();
+        // Open the submenu
```
Comment (shayan-deriv) [inline src/components/AppLayout/Submenu /__tests__/Submenu.spec.tsx:28]: do we need this? 🤔

File: src/components/AppLayout/Submenu /index.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            <button
+                className={clsx("submenu_button", className)}
+                onClick={onOpenSubmenu}
+            >
+                <span className="submenu_wrapper">
+                    {icon}
+                    <Text size={labelSize}>{label}</Text>
+                </span>
+                <LegacyChevronRight1pxIcon iconSize="xs" />
+            </button>
```
Comment (shayan-deriv) [inline src/components/AppLayout/Submenu /index.tsx:66]: why do we have this? 
it should be a isolated component controlled by the consumer using `isOpen` prop

File: src/components/AppLayout/Submenu /index.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+ * @example
+ * // To trigger an exit animation, change isOpen to false:
+ * <Submenu isOpen={false} className="custom-submenu">
+ *   <p>Menu Content Here</p>
+ * </Submenu>
+ */
+export const Submenu = ({
+    children,
+    className,
+    isOpen,
```
Comment (shayan-deriv) [inline src/components/AppLayout/Submenu /index.tsx:42]:
```suggestion
    isOpen,
    ...rest
```

File: src/components/AppLayout/Submenu /index.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+        return () => {
+            if (timerRef.current) clearTimeout(timerRef.current);
+        };
+    }, [isOpen]);
+
+    if (!isMounted) return null;
+    return (
+        <div
+            className={clsx("submenu", { submenu_exit: isClosing }, className)}
```
Comment (shayan-deriv) [inline src/components/AppLayout/Submenu /index.tsx:68]:
```suggestion
            className={clsx("submenu", { submenu_exit: isClosing }, className)} {...rest}
```

**Review by farrah-deriv — APPROVED** [review]

File: src/components/AppLayout/Submenu /index.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+ */
+export const Submenu = ({
+    children,
+    className,
+    isOpen,
+    ...rest
+}: PropsWithChildren<TSubmenu>) => {
+    const timerRef = useRef<ReturnType<typeof setInterval> | null>(null);
+    const [isMounted, setIsMounted] = useState(isOpen);
+    const [isClosing, setIsClosing] = useState(false);
```
Comment (shayan-deriv) [inline src/components/AppLayout/Submenu /index.tsx:47]: an reason for having two states ?
one should be enough.just like the [drawer](https://github.com/deriv-com/ui/blob/main/src/components/AppLayout/Drawer/index.tsx) component.

Comment (niloofar-deriv) [reply src/components/AppLayout/Submenu /index.tsx]: No need for that, thanks for noticing this. I'll remove it 🙏

**Review by shayan-deriv — APPROVED** [review]

**Pull request merged**